solver/src/driver.rs

```diff
@@ -314,14 +337,24 @@ struct RatedSettlement {
     objective_value: BigRational,
 }
 
+// Goes through the settlements in order and tries to merge at most MAX_SETTLEMENT_MERGES together.
+fn merge_settlements(mut settlements: impl Iterator<Item = Settlement>) -> Result<Settlement> {
+    let mut merged = settlements
+        .next()
+        .ok_or_else(|| anyhow!("no settlements"))?;
+    for settlement in settlements.take(MAX_MERGED_SETTLEMENTS - 1) {
+        merged = merged.merge(settlement)?;
```

Comment:
If a specific settlement cannot be merged into the existing bunch, shouldn't we still try to merge the rest of the settlements? This way we can still reduce the number of total transactions we are waiting for.

Reply:
Fixed.